flexmeasures_openweathermap/utils/locating.py

```diff
@@ -108,13 +108,21 @@ def find_weather_sensor_by_location_or_fail(
     return weather_sensor
 
 
-def get_asset_id_location(asset_id: int) -> List[Tuple[float, float]]:
+def get_location_by_asset_id(asset_id: int) -> Tuple[float, float]:
     """Get location for forecasting by passing an asset id"""
     asset = GenericAsset.query.filter(
         GenericAsset.generic_asset_type_id == asset_id
     ).one_or_none()
-    if asset is None:
+    if asset is not None:
+        asset_type = GenericAssetType.query.filter(
```

Comment:
No need for an extra lookup.
With SQLAlchemy that happens under the hood, and you can simply access `asset.generic_asset_type`.